src/acl.c

```diff
@@ -1043,9 +1061,24 @@ const char *ACLSetUserStringError(void) {
  * lifetime. */
 void ACLInitDefaultUser(void) {
     DefaultUser = ACLCreateUser("default",7);
+    ACLSetDefaultUserState();
+}
+
+/*
+ * Set's the base state of DefaultUser.
+ */
+void ACLSetDefaultUserState(void) {
     ACLSetUser(DefaultUser,"+@all",-1);
-    ACLSetUser(DefaultUser,"~*",-1);
-    ACLSetUser(DefaultUser,"&*",-1);
+    ACLSetUser(DefaultUser, "~*", -1);
+    /*
+     * Based on the acl_pubsub_default, default access
+     * is set to allchannels or nochannels.
+     */
+    if (server.acl_pubsub_default & USER_FLAG_ALLCHANNELS) {
```

Comment:
We probably don't want to change the behavior here. The DefaultUser, paradoxically, doesn't have the same permission as new users. The DefaultUser is intended to mimic the behavior of the pre-Redis 6 user.

Reply:
Ok, this might be a bit strange behaviour.